packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -163,6 +163,8 @@ function h(sel: string, data: VElementData, children: VNodes = EmptyArray): VEle
         });
     }
 
+    data = applyTemporaryCompilerV5SlotFix(data);
```

Comment:
Let's add comments to these lines similar to the ones in the other files.
```javascript
// TODO [#3974]: remove temporary logic to support v5 compiler + v6+ engine
```

Reply:
https://github.com/salesforce/lwc/pull/3973/commits/83d9c896d8708906678f568cfe3eacdde2114e03